storages/jsonl-storage/src/lib.rs

```diff
@@ -296,26 +289,33 @@ impl StoreMut for JsonlStorage {
             .open(table_path)
             .map_storage_err()?;
 
-        let labels = schema
-            .column_defs
-            .unwrap_or_default()
-            .into_iter()
-            .map(|column_def| column_def.name);
+        let labels = schema.column_defs.unwrap_or_default();
+        let labels = labels
+            .iter()
+            .map(|column_def| column_def.name.as_str())
+            .collect::<Vec<_>>();
 
         for row in rows {
             let json_string = match row {
-                DataRow::Map(hash_map) => JsonIter::Map(hash_map.into_iter()),
-                DataRow::Vec(values) => JsonIter::Vec(labels.clone().zip(values.into_iter())),
-            }
-            .map(|(key, value)| {
-                let value = JsonValue::try_from(value)?.to_string();
-
-                Ok(format!("\"{key}\": {value}"))
-            })
-            .collect::<Result<Vec<_>>>()?
-            .join(", ");
-
-            writeln!(file, "{{{json_string}}}").map_storage_err()?;
+                DataRow::Vec(values) => labels
+                    .iter()
+                    .zip(values.into_iter())
+                    .map(|(key, value)| {
+                        let value = JsonValue::try_from(value)?.to_string();
+                        Ok(format!("\"{}\": {}", key, value))
+                    })
+                    .collect::<Result<Vec<_>>>()?
+                    .join(", "),
+                DataRow::Map(hash_map) => hash_map
+                    .into_iter()
+                    .map(|(key, value)| {
+                        let value = JsonValue::try_from(value)?.to_string();
+                        Ok(format!("\"{}\": {}", key, value))
+                    })
+                    .collect::<Result<Vec<_>>>()?
+                    .join(", "),
```

Comment:
Oh, I just recognized this.
It looks that it is a kind of custom serialization from HashMap to JsonString.

```rust
let value = JsonValue::try_from(value)?.to_string();
Ok(format!("\"{}\": {}", key, value))
```
Rather than the above,
how about making JsonValue::Object and serialize as string rather than using `format` macro?